assets/source/setup-guide/app/components/TermsAndConditionsModal.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	Modal,
+	__experimentalText as Text, // eslint-disable-line @wordpress/no-unsafe-wp-apis --- _experimentalText unlikely to change/disappear and also used by WC Core
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import documentationLinkProps from '../helpers/documentation-link-props';
+
+const tosHref = 'https://business.pinterest.com/business-terms-of-service/';
+const privacyPolicyHref = 'https://policy.pinterest.com/privacy-policy';
+const advertisingServicesAgreementHref =
+	'https://business.pinterest.com/pinterest-advertising-services-agreement/';
+
+/**
+ * Modal used for displaying terms and conditions information required for Ads Credit Feature.
+ *
+ * @param {Function} onModalClose Action to call when the modal gets closed.
+ *
+ * @fires wcadmin_pfw_documentation_link_click with `{ link_id: 'terms-of-service', context: 'ads-credits-terms-and-conditions' }`
+ * @fires wcadmin_pfw_documentation_link_click with `{ link_id: 'privacy-policy', context: 'ads-credits-terms-and-conditions' }`
+ * @fires wcadmin_pfw_documentation_link_click with `{ link_id: 'advertising-services-agreement', context: 'ads-credits-terms-and-conditions' }`
+ *
+ * @return {JSX.Element} Rendered element.
+ */
+const AdsCreditsTermsAndConditionsModal = ( { onModalClose } ) => {
+	return (
+		<Modal
+			title={
+				<>{ __( 'Terms & Conditions', 'pinterest-for-woocommerce' ) }</>
+			}
+			onRequestClose={ onModalClose }
+			className="pinterest-for-woocommerce-landing-page__credits-section__tac-modal"
+		>
+			<Text variant="body">
```

Comment:
I'd prefer to use directly `Text` and `Link` elements instead of nesting `div`, `p` and `a` tags inside a text block